tests/test_ec2/test_elastic_block_store.py

```diff
@@ -561,6 +561,11 @@ def test_volume_attach_and_detach_boto3():
     ec2 = boto3.resource("ec2", region_name="us-east-1")
     reservation = client.run_instances(ImageId=EXAMPLE_AMI_ID, MinCount=1, MaxCount=1)
     instance = reservation["Instances"][0]
+
+    volumes = client.describe_volumes()["Volumes"]
```

Comment:
EC2 tests run in parallel in ServerMode, so this call could also return volumes from other tests.
You can use `describe_volumes(VolumeIds=[..])` to only get the volume that you need.

(See http://docs.getmoto.org/en/latest/docs/contributing/development_tips/tests.html for more info)

Reply:
Good catch there. 

This didn't work in my case because I don't have a volume id to filter by (run_instances doesn't return a volume id).

Fortunately `describe_volumes` also allows Filters that support matching by instance id :+1: